src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/PolymorphicTests.cs

```diff
@@ -529,6 +529,124 @@ public async Task AnonymousType()
             Assert.Equal(Expected, json);
         }
 
+        [Fact]
```

Comment:
Should be moved to PolymorphicTests.CustomTypeHierarchies.cs which specifically test this type of polymorphism.